rust/src/pgsql/pgsql.rs

```diff
@@ -288,6 +297,23 @@ impl PgsqlState {
 
                 // Important to keep in mind that: "In simple Query mode, the format of retrieved values is always text, except when the given command is a FETCH from a cursor declared with the BINARY option. In that case, the retrieved values are in binary format. The format codes given in the RowDescription message tell which format is being used." (from pgsql official documentation)
             }
+            PgsqlFEMessage::ConsolidatedCopyDataIn(_) => {
+                match self.get_curr_state() {
+                    PgsqlStateProgress::FirstCopyDataInReceived
+                    | PgsqlStateProgress::ConsolidatingCopyDataIn => {
+                        return Some(PgsqlStateProgress::ConsolidatingCopyDataIn);
+                    }
+                    PgsqlStateProgress::CopyInResponseReceived => {
+                        return Some(PgsqlStateProgress::FirstCopyDataInReceived);
+                    }
+                    _ => {
+                        // TODO should we return None here, since it's an unexpected state?
```

Comment:
Question to better understand what makes more sense or not.

From a state machine perspective, I think that `None` or something like an invalid state should be the case.

From a network inspection perspective, I guess we should parse what we see, regardless of how much it respects the RFC?

Reply:
We are in a network inspection perspective, so we should be relax, and try to fall back to the best state

Reply:
Thanks, I'll see how to convey that.